src/controllers/hall.controllers.ts

```diff
@@ -0,0 +1,71 @@
+import { NextFunction, Request, Response } from 'express';
+import { Service } from 'typedi';
+
+import {
+  createHallService,
+  getHallByIdService,
+  getAllHallService,
+  updateHallService,
+  deleteHallService,
+} from '../services/hall.services';
+
+import { CustomError } from '../utils/response/custom-error/customError';
+
+@Service()
+class HallControllers {
+  constructor() {}
+  createHall = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const newHall = await createHallService(req.body, next);
+      if (newHall != null) {
+        res.customSuccess(201, 'Hall created successfully', newHall);
+      }
+    } catch (error) {
+      next(error);
+    }
+  };
+  getHallById = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const hall = await getHallByIdService(req.params.id, next);
+      if (hall != null) {
+        res.customSuccess(200, `hall found`, hall);
+      }
+    } catch (error) {
+      next(error);
+    }
+  };
+  getAllHalls = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const query = req.query;
+      const option = req.query;
+      const halls = await getAllHallService(query, option, next);
+      if (halls != null) {
+        res.customSuccess(200, 'halls found', halls);
+      }
```

Comment:
Destructure the req.query bases on the things required like the limit, pages e.t.c. You can check what i'm saying on the the member/worker's controller's file.